app/controllers/application_controller.rb

```diff
@@ -7,3 +7,18 @@ def current_user
 
   helper_method :current_user
 end
+
+class ApplicationController < ActionController::Base
+  protect_from_forgery with: :exception
+
+  before_action :configure_permitted_parameters, if: :devise_controller?
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: [:name])
+  end
+
+  def configure_devise_parameters
+    devise_parameter_sanitizer.permit(:sign_up) do |user|
+      user.permit(:name, :email, :password, :password_confirmation)
+    end
+  end
+end
```

Comment:
- You have done well so far, but you have created a base controller inside a base controller, I suggest that you have just one base controller and ensure your application works correctly, please fix this 👍 👍 
- Also in the parent controller, according to the project requirements, you are required to remove the `current_user` method as this is already handled by `devise`. Please fix this 😄 
- Please also ensure that you authenticate all users before any action is accessed, please fix this 😄

Reply:
done